packages/api/src/controllers/project.ts

```diff
@@ -177,4 +181,115 @@ app.delete("/:id", authorizer({}), async (req, res) => {
   res.end();
 });
 
+app.post(
+  "/job/projects-cleanup",
+  authorizer({ anyAdmin: true }),
+  async (req, res) => {
+    // import the job dynamically to avoid circular dependencies
+    const { default: projectsCleanup } = await import(
+      "../jobs/projects-cleanup"
+    );
+
+    const limit = parseInt(req.query.limit?.toString()) || 1000;
+
+    const { cleanedUp } = await projectsCleanup(
+      {
+        ...req.config,
+        projectsCleanupLimit: limit,
+      },
+      req,
+      { jobsDb },
+    );
+
+    res.status(200);
+    res.json({ cleanedUp });
+  },
+);
+
+export function triggerCleanUpProjectsJob(
+  projects: Project[],
+  req: Request,
+): [Project[], Promise<void>] {
+  if (!projects.length) {
+    return [projects, Promise.resolve()];
+  }
+
+  const jobPromise = Promise.resolve().then(async () => {
+    try {
+      await Promise.all(projects.map((s) => cleanUpProject(jobsDb, s, req)));
+    } catch (err) {
+      const ids = projects.map((s) => s.id);
+      logger.error(`Error cleaning up projectId=${ids} err=`, err);
+    }
+  });
+
+  return [projects, jobPromise];
+}
+
+async function cleanUpProject(db: DB, project: Project, req: Request) {
+  let [assets] = await db.asset.find({
+    filters: {
+      projectId: project.id,
+      deleted: false,
+    },
+  });
+
+  let [streams] = await db.stream.find({
+    filters: {
+      projectId: project.id,
+      deleted: false,
+    },
+  });
+
+  let [signingKeys] = await db.signingKey.find({
+    filters: {
+      projectId: project.id,
+      deleted: false,
+    },
+  });
+
+  let [webhooks] = await db.webhook.find({
+    filters: {
+      projectId: project.id,
+      deleted: false,
+    },
+  });
+
+  let [sessions] = await db.session.find({
+    filters: {
+      projectId: project.id,
+      deleted: false,
+    },
+  });
+
+  for (const asset of assets) {
+    await req.taskScheduler.deleteAsset(asset.id);
+  }
+
+  for (const stream of streams) {
+    await db.stream.update(stream.id, {
```

Comment:
FWIW there's a `markDeleted` function on the tables which is generally used for the soft deletion